front/assets/js/organization_okta/index.tsx

```diff
@@ -0,0 +1,230 @@
+import { render, createContext } from "preact";
+import { useState, useContext, useEffect } from "preact/hooks";
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { Notice } from "js/notice";
+
+interface group {
+  id: string;
+  name: string;
+}
+
+interface role {
+  id: string;
+  name: string;
+}
+
+export class State {
+  groups: group[];
+  roles: role[];
+  defaultRoleId: string;
+  saveUrl: string;
+  cancelUrl: string;
+  mappings?: any[];
```

Comment:
I think this field does not need to be optional if we can provide [a default value](https://github.com/semaphoreio/semaphore/pull/109/files#diff-bed9202c3cc1ab091187724947cd02d81f95dff8ce1ef1de80a07603273e7b1eR46).